### ISSUE TITLE
styleguide - error when default rich text feature uses gettext_lazy description

### ISSUE THREAD

**Issue by enzedonline** [wagtail/wagtail#11894]
### Issue Summary

If the default feature set for WAGTAILADMIN_RICH_TEXT_EDITORS includes a feature with a translatable `gettext_lazy` description, the styleguide throws a 500 error trying to render the default rich text editor in the form fields ([ref](https://github.com/wagtail/wagtail/blob/9dc38076ed9d12d63dc40258e6135b43d4054d96/wagtail/contrib/styleguide/templates/wagtailstyleguide/base.html#L434)):
```django
<li>{% formattedfield field %}</li>
```
`TypeError: Object of type __proxy__ is not JSON serializable`

### Steps to Reproduce

Use the [example](https://docs.wagtail.org/en/stable/extending/extending_draftail.html#formatting-extensions) from docs to register a feature, except defining the description with a `gettext_lazy` call:

```python
from django.utils.translation import gettext_lazy as _
import wagtail.admin.rich_text.editors.draftail.features as draftail_features
from wagtail.admin.rich_text.converters.html_to_contentstate import InlineStyleElementHandler
from wagtail import hooks

@hooks.register('register_rich_text_features')
def register_mark_feature(features):
    feature_name = 'mark'
    type_ = 'MARK'
    tag = 'mark'
    control = {
        'type': type_,
        'label': '☆',
        'description': _('Mark'), # <-- use gettext_lazy() here for translatable description
    }
    features.register_editor_plugin(
        'draftail', feature_name, draftail_features.InlineStyleFeature(control)
    )
    db_conversion = {
        'from_database_format': {tag: InlineStyleElementHandler(type_)},
        'to_database_format': {'style_map': {type_: tag}},
    }
    features.register_converter_rule('contentstate', feature_name, db_conversion)
    features.default_features.append('mark')
```
In settings, define a custom default feature set including the custom block:

```python
WAGTAILADMIN_RICH_TEXT_EDITORS = {
    'default': {
        'WIDGET': 'wagtail.admin.rich_text.DraftailRichTextArea',
        'OPTIONS': {
            'features': ['h2', 'h3', 'h4', 'bold', 'italic', 'underline', 'mark' ]
        }
    },
}
```

-   I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: yes

### Technical details

-   Python version: 3.12
-   Django version: 4.2.7
-   Wagtail version: 6.02
-   Browser version: Chrome 124

**Comment by lb-** [comment]
Related to https://github.com/wagtail/wagtail/pull/11620

I'm not sure if that PR considered lazy values coming in via the inline style filter usage. 

Thanks for raising @enzedonline

**Comment by laymonage** [comment]
Thanks for the report!

I can confirm this issue in `stable/6.0.x`, but not the `stable/6.1.x` branch. Looking at #11620 it seems the addition of the custom encoder in https://github.com/wagtail/wagtail/pull/11620/files#diff-52405caa6db383bd6f5b707eba45b66df9d0a7ca8ecb3d8bb1467a24f0c34ad7R85-R87 fixed the issue.

I can also confirm this issue in `stable/5.2.x`, so I think this has been around for a while and not a regression. The fix will be included in the upcoming 6.1 release, but not in 6.0.3.

That said, it looks like the approach in #11620 could be simplified using `django.core.serializers.json.DjangoJSONEncoder` instead of a custom encoder implementation.
